File: Module 3/userService.js
import api from './api';

// Add a new user
export const addUser = (userData) => api.post('/users/add', userData);

// Get all users
export const getAllUsers = () => api.get('/users/all');

// Get user by email
export const getUserByEmail = (email) => api.get(`/users/${email}`);

File: Final Project/backend/backend/src/main/java/com/projectmanagementapp/backend/service/UserService.java
package com.projectmanagementapp.backend.service;

import com.projectmanagementapp.backend.model.User;
import com.projectmanagementapp.backend.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    // Save new user
    public User saveUser(User user) {
        return userRepository.save(user);
    }

    // Get all users
    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    // Get user by email
    public Optional<User> getUserByEmail(String email) {
        return userRepository.findByEmail(email);
    }
}
